src/components/Charts/Charts.vue

```diff
@@ -0,0 +1,177 @@
+<template>
+  <v-chart class="chart" :option="option" />
+</template>
+
+<script>
+import { use } from "echarts/core";
+import { CanvasRenderer } from "echarts/renderers";
+import { BarChart } from "echarts/charts";
+import {
+  TitleComponent,
+  TooltipComponent,
+  LegendComponent,
+} from "echarts/components";
+import VChart, { THEME_KEY } from "vue-echarts";
+import { ref, defineComponent } from "vue";
+import { ToolboxComponent } from "echarts/components";
+import { GridComponent } from "echarts/components";
+import { LineChart } from "echarts/charts";
+
+use([
+  CanvasRenderer,
+  BarChart,
+  TitleComponent,
+  TooltipComponent,
+  LegendComponent,
+  ToolboxComponent,
+  GridComponent,
+  LineChart,
+]);
+
+export default defineComponent({
+  name: "Charts",
+  components: {
+    VChart,
+  },
+  provide: {
+    [THEME_KEY]: "light",
+  },
+  setup() {
+    const app = {};
+    const posList = [
+      "left",
+      "right",
+      "top",
+      "bottom",
+      "inside",
+      "insideTop",
+      "insideLeft",
+      "insideRight",
+      "insideBottom",
+      "insideTopLeft",
+      "insideTopRight",
+      "insideBottomLeft",
+      "insideBottomRight",
+    ];
+
+    app.configParameters = {
+      rotate: {
+        min: -90,
+        max: 90,
+      },
+      align: {
+        options: {
+          left: "left",
+          center: "center",
+          right: "right",
+        },
+      },
+      verticalAlign: {
+        options: {
+          top: "top",
+          middle: "middle",
+          bottom: "bottom",
+        },
+      },
+      position: {
+        options: posList.reduce(function (map, pos) {
+          map[pos] = pos;
+          return map;
+        }, {}),
+      },
+      distance: {
+        min: 0,
+        max: 100,
+      },
+    };
+
+    app.config = {
+      rotate: 86,
+      align: "left",
+      verticalAlign: "bottom",
+      position: "insideBottom",
+      distance: 35,
+    };
+    const labelOption = {
+      show: true,
+      position: app.config.position,
+      distance: app.config.distance,
+      align: app.config.align,
+      verticalAlign: app.config.verticalAlign,
+      rotate: app.config.rotate,
+      formatter: "{c}  {name|{a}}",
+      fontSize: "2rem",
+      rich: {
+        name: {},
+      },
+    };
+    const option = ref({
```

Comment:
nie dajemy obiektów do refów

Reply:
Jeśli nie nadajecie niczego dynamicznie, to nie musi być ref